src/internal/collection/README.md

```diff
@@ -0,0 +1,103 @@
+# Collections
+
+Collections are a library for storing protobufs in a persistent database with a simple interface for reading, writing, and watching for changes.  There are two existing implementations, one for etcd and one for postgres.
+
+A collection can be constructed via `NewEtcdCollection` or `NewPostgresCollection`, depending on the backend to be used.  This function must be given:
+ * a template protobuf `Message`, which will be clones for use in serialization and deserialization
+ * a set of fields to create secondary indexes for (used by `GetByIndex` and `WatchByIndex`)
+
+## API
+
+The collection API is split into two means of accessing a collection - `ReadOnly` and `ReadWrite`.  A `ReadOnly` collection does not make any consistency guarantees, while a `ReadWrite` collection is guaranteed to do all reads and writes within a single transaction, although this comes with trade-offs.
```

Comment:
So not even ReadCommitted? I could be reading data from transactions in progress?

Reply:
If we want identical semantics to etcd, we should use `Serializable` - right now it's just using the default of `ReadCommitted`, but I think we should probably move it to `ReadRepeatable` - I don't think we need to guarantees of `Serializable` in postgres.